old-branches.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/bash
+
+timerange=$1
+
+out=''
+branches=()
+while read -r line; do
+  elems=($line)
+  name=${elems[1]#"refs/heads/"}
+  if [[ ${name} == release* ]]; then
+    continue
+  fi
+  hash=${elems[0]}
+  date=$(git show -s --format="%cs" ${hash})
+  if [[ ${date} != ${timerange}* ]]; then
+    continue
+  fi
+  out+=$(printf "%s:%s" ${date} ${name})
+  out+="\n"
+  branches+=(${name})
+done <<< $(git ls-remote -h origin)
+
+echo "Going to delete the following branches:"
+echo
+printf "${out}" | column -t -s ":" | sort
```

Comment:
As a general practice, I would use `echo` here rather than `printf`, because `printf` would do something special if there were formatting characters in `${out}`.

Reply:
I couldn't figure out how to get `echo` to work here: it prints literal `\n`s instead of actually breaking the text up by newlines. I feel like I tried a dozen different things to get it to not escape those newlines and using `printf` was the only one that worked for me.

Reply:
Ah, I think I see the problem. `echo` is not escaping the newlines - you are adding escaped newlines to `$out` on line 19. However, `printf` sees a backslash-escaped newline and turns it into a real newline.

According to
https://catonmat.net/bash-one-liners-explained-part-two#:~:text=%24'%5Cn'%20is%20bash%20idiomatic,z%2C%20and%20the%20newline%20character.&text=This%20one%2Dliner%20uses%20command,the%20command%20with%20its%20output, the bash idiomatic way to get a newline is `$'\n'`, so:

```
out+=$'\n'
```

Reply:
`$'\n'` really? *siiigh*

Thanks!

Reply:
Yeah, same! I just learned about this now. My go-to would have been `out+="$(echo -e '\n')`, which is ugly but at least I'm used to it. 😃